svetlana_kurakina_ndvi.py

```diff
@@ -0,0 +1,904 @@
+
+# YOUR ANSWER HERE
+# 1. I would plan flights for HARV site starting from mid May and until the first week of October.
+# 2. Flights for SJER would have to be planned for a short window between mid February and end of March.
+
+# # Question 2 (10 points)
+# 
+# How could you modify your workflow to look at vegetation changes over time in each site? 
+# 
+# Answer the question in 2-3 sentences in the Markdown cell below.
+
+# YOUR ANSWER HERE:
+# 
+# 1. I would probably ended up making separate dataframes for each site.
+# 2. I think, it may be good idea to process more landsat scenes (for few years). I can see seasonal changes on the data from one year, but long-term trend would be interesting to explore.
+# 3. Would be interesting to compare NDVI pattern with tempreture and precipitation variations across years.
+
+# # Do not edit this cell! (10 points)
+# 
+# The notebook includes:
+# * additional Markdown cells throughout the notebook to describe: 
+#     * the data that you used - and where it is from
+#     * how data are being processing
+#     * how the code is optimized to run fast and be more concise
+
+# # Do not edit this cell! (20 points)
+# 
+# The notebook will also be checked for overall clean code requirements as specified at the **top** of this notebook. Some of these requirements include (review the top cells for more specifics): 
+# 
+# * Notebook begins at cell [1] and runs on any machine in its entirety.
+# * PEP 8 format is applied throughout (including lengths of comment and code lines).
+# * No additional code or imports in the notebook that is not needed for the workflow.
+# * Notebook is fully reproducible. This means:
+#    * reproducible paths using the os module.
+#    * data downloaded using code in the notebook.
+#    * all imports at top of notebook.
+
+# ## BONUS - Export a  .CSV File to Share (10 points possible)
+# 
+# This is optional - if you export a **.csv** file with the columns specified above: Site, Date and NDVI Value you can get an additional 10 points.
+# 
+# * FULL CREDIT: File exists in csv format and contains the columns specified.
+# We will check your github repo for this file!
+# 
+
+# In[14]:
+
+
+# Export final datafarame to csv stored in the outputs directory
+
+# Check if output folder exists
+# Create output folder if necessary
+
+ndvi_output = os.path.join(
+    'ndvi-automation', 
+    'outputs'
+)
+    
+if os.path.exists(ndvi_output):
+    print('Output directory exists')
+else:
+    print('Output directory does not exist but it is being created.')
+    os.makedirs(ndvi_output)
+
+# Create path and filename
+ndvi_output_csv = os.path.join(
+    ndvi_output, 
+    'ndvi_output.csv'
+)
+
+# Export final df to csv
+ndvi_clean.to_csv(ndvi_output_csv)
+
+# Check that output csv exists
+if os.path.exists(ndvi_output_csv):
+    print('Output CSV exists')
+else:
+    print('Oh no, final CSV does not exist in the outputs directory.')
```

Comment:
Really thorough to ensure that the paths and output folders exist.

Reply:
It is probably unnessesary here, but I liked this approach when we did something like that in one of the earlier homeworks...